src/ocamlorg_web/lib/handler.ml

```diff
@@ -637,7 +637,28 @@ let packages_search t req =
     Dream.from_percent_encoded
       (match Dream.query req "q" with Some search -> search | None -> "")
   in
-  let results = List.map (Package_helper.frontend_package t) current_items in
+
+  let open Lwt.Syntax in
+  let documentation_status (pkg : Ocamlorg_package.t) =
+    let* package_documentation_status =
+      Ocamlorg_package.documentation_status ~kind:`Package t pkg
+    in
+
+    Lwt.return
+      (match package_documentation_status with
+      | Some { failed = false; _ } -> Ocamlorg_frontend.Package.Success
+      | Some { failed = true; _ } -> Failure
+      | None -> Unknown)
+  in
+
+  let* results =
+    Lwt_list.map_p
+      (fun pkg ->
+        let+ doc_status = documentation_status pkg in
+        (Package_helper.frontend_package t pkg, doc_status))
+      current_items
+  in
+
```

Comment:
We should probably make documentation status part of `Ocamlorg_frontend.Package` so that `Package_helper.frontend_package` can deal with the lookup of documentation status. :thinking:

Reply:
I'm not sure I get this. Do you mean to move this function to the `Ocamlorg_frontend.Package`?